### PR TITLE
test: Frontend Vitest Test-Utils Module

### PR DESCRIPTION
**Pull request by sahil-noon**
## Meta

| Change ID | Type | Confidence | Plan | Review |
|-----------|------|------------|------|--------|
| `5hm6` | test | 4.9/5.0 | 11/11 tasks, 11/11 acceptance ✓ | ✓ 1 cycle |

| Impact | +/− | Net |
|--------|----:|----:|
| raw | +420 / −276 | +144 |
| **true** | **+113 / −276** | **-163** |
| └ impl  (clamped from net -131) | +108 / −239 | +0 |
| └ tests | +5 / −37 | -32 |

<sub>excludes `fab/`, `docs/` · generated by fab-kit v2.16.10</sub>

**Pipeline:** [intake](https://github.com/sahil87/run-kit/blob/260731-5hm6-frontend-vitest-test-utils/fab/changes/260731-5hm6-frontend-vitest-test-utils/intake.md) ✓ → [apply](https://github.com/sahil87/run-kit/blob/260731-5hm6-frontend-vitest-test-utils/fab/changes/260731-5hm6-frontend-vitest-test-utils/plan.md) ✓ → review ✓ → hydrate ✓ → ship → review-pr

## Summary

Nine near-identical fixture factories (`makeWindow` x5, `makeSession`, `session()`, `baseSessions`, `makeWindowWithPanes`) and ~11 predicate-style `matchMedia` stub definitions were copy-pasted across `app/frontend/src` test files, causing drift in trivial defaults and forcing mechanical multi-file edits whenever `WindowInfo` gains a field. A shared, imported-per-test `test-utils` module (not a global `setupFiles` install) consolidates this scaffolding while preserving each test file's explicit control over its environment.

## Changes

- New module: `app/frontend/src/test-utils/` (`fixtures.ts` with canonical `makeWindow`/`makeSession`/`makeWindowWithPanes` factories, `match-media.ts` with a `stubMatchMedia(predicate?)` helper)
- Migration of existing test files: 17 test files updated to import the shared helpers and drop their local duplicated copies
- Explicit non-goals: the `mockReturnValue(mql)` controllable-MQL matchMedia family is left as-is (different shape, real event-driven behavior); no production code touched; no global auto-install via `setupFiles`

### CODE REVIEW

File: fab/changes/260731-5hm6-frontend-vitest-test-utils/plan.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+- [x] A-007 R5: `just test-frontend` passes with the same counts as the pre-change baseline (114 files / 2039 tests) — re-run at review: **114 files / 2039 tests passed**, exact baseline match
+- [x] A-008 R5: `npx tsc --noEmit` in `app/frontend` exits 0 — re-run at review: exit 0
+
+### Edge Cases & Error Handling
+
+- [x] A-009 R5: `git diff --name-only` shows only test files plus the new `src/test-utils/` — zero production files, no `vitest.config.ts`/`test-setup.ts` edits, `mockReturnValue(mql)` family untouched — verified: 17 modified files all `*.test.ts(x)` + 2 new `test-utils/*.ts`; `vitest.config.ts`/`test-setup.ts` show no diff; use-coarse-pointer, theme-context, theme-selector, swatch-popover, terminal-client, sidebar.test.tsx all UNTOUCHED; `window-row.test.tsx:20` `mockMatchMedia` and `compose-strip.test.tsx:98` `mockReturnValue` preserved in-place
```
[the remark this answers is not in the document]
Comment (sahil-noon) [reply fab/changes/260731-5hm6-frontend-vitest-test-utils/plan.md]: Fixed — scoped the A-009 acceptance bullet to `app/frontend/` and noted that fab pipeline bookkeeping (`fab/changes/`) and the `docs/memory/` hydrate artifact sit outside the tests-only constraint. (54b1915)

File: app/frontend/src/test-utils/fixtures.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+/** Build a `ProjectSession` with one canonical window. */
+export function makeSession(overrides: Partial<ProjectSession> = {}): ProjectSession {
+  return {
+    name: "alpha",
+    windows: [makeWindow()],
+    ...overrides,
+  };
+}
```
[the remark this answers is not in the document]
Comment (sahil-noon) [reply app/frontend/src/test-utils/fixtures.ts:28]: Fixed — `makeSession` now computes `windows: overrides.windows ?? [makeWindow()]`, so the default window is only built when the caller doesn't provide one. (54b1915)
